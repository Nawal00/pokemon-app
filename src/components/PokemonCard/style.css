.pokemon-card-wrapper {
  text-align: center;
  flex-basis: 50%;
}

.pokemon-card {
  background-color: #a0d4e4;
  border-radius: 5px;
  padding: 1rem;
}

.pokemon-card > div {
  position: relative;
}

.pokemon-card .close-btn {
  position: absolute;
  cursor: pointer;
  height: 30px;
  width: 30px;
  top: -20px;
  right: 0px;
}

.pokemon-card button {
  background-color: #4865c2;
  border-radius: 5px;
  margin-left: 5px;
  padding: 5px;
  color: #ffffff;
}
