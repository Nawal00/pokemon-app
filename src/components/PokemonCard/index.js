import PropTypes from "prop-types";
import "./style.css";

const PokemonCard = ({
  name,
  height,
  weight,
  abilities,
  types,
  sprites,
  setPokemonInfo,
  loading,
}) => {
  if (loading) return <p>Loading...</p>;

  return (
    <div className="pokemon-card-wrapper" data-testid="pokemon-card-wrapper">
      <div className="pokemon-card">
        <div>
          <h2>{name.toUpperCase()}</h2>
          <button
            className="close-btn"
            data-testid="close-btn"
            onClick={() => setPokemonInfo({})}
          >
            x
          </button>
        </div>
        <img
          src={sprites.other.dream_world.front_default}
          alt="a pokemon character"
        />
        <p>
          <b>Height:</b> {height} (decimetre)
        </p>
        <p>
          <b>Weight:</b> {weight} (hectogram)
        </p>
        <p>
          <b>Abilities:</b>
          {abilities.map(({ ability }) => (
            <button key={ability.name}>{ability.name}</button>
          ))}
        </p>
        <p>
          <b>Types:</b>
          {types.map(({ type }) => (
            <button key={type.name}>{type.name}</button>
          ))}
        </p>
      </div>
    </div>
  );
};

export default PokemonCard;

PokemonCard.propTypes = {
  name: PropTypes.string.isRequired,
  height: PropTypes.number,
  weight: PropTypes.number,
  abilities: PropTypes.arrayOf(
    PropTypes.shape({
      ability: PropTypes.shape({
        name: PropTypes.string.isRequired,
        url: PropTypes.string,
      }),
    })
  ),
  types: PropTypes.arrayOf(
    PropTypes.shape({
      type: PropTypes.shape({
        name: PropTypes.string.isRequired,
        url: PropTypes.string,
      }),
    })
  ),
  sprites: PropTypes.shape({
    other: PropTypes.shape({
      dream_world: PropTypes.shape({
        front_default: PropTypes.string,
      }),
    }),
  }),
  setPokemonInfo: PropTypes.func.isRequired,
  loading: PropTypes.bool,
};
