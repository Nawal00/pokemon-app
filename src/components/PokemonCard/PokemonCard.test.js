import { render, screen, fireEvent } from "@testing-library/react";
import PokemonCard from "./";

const props = {
  name: "pikachu",
  height: 30,
  weight: 99,
  abilities: [{ ability: { name: "solar-flaire" } }],
  types: [{ type: { name: "wild" } }],
  sprites: {
    other: {
      dream_world: {
        front_default: "/test.svg",
      },
    },
  },
  setPokemonInfo: jest.fn(),
  loading: false,
};

describe("Given the `PokemonCard` component", () => {
  it("should render with correct props", () => {
    const { container } = render(<PokemonCard {...props} />);

    expect(container).toMatchSnapshot();
  });

  describe("When user clicks on close `x` button", () => {
    it("should call `setPokemonInfo` with correct params", () => {
      render(<PokemonCard {...props} />);

      const closeButton = screen.getByTestId("close-btn");

      fireEvent.click(closeButton);

      expect(props.setPokemonInfo).toHaveBeenCalledWith({});
    });
  });

  describe("When the `loading` prop is set to true", () => {
    it("should display a loading text", () => {
      render(<PokemonCard {...props} loading />);

      const loading = screen.getByText("Loading...");

      expect(loading).toBeInTheDocument();
    });
  });
});
