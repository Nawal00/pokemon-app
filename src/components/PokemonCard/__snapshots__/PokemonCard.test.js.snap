// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the \`PokemonCard\` component should render successfully 1`] = `
<div>
  <div
    class="pokemon-card-wrapper"
    data-testid="pokemon-card-wrapper"
  >
    <div
      class="pokemon-card"
    >
      <div>
        <h2>
          PIKACHU
        </h2>
        <button
          class="close-btn"
          data-testid="close-btn"
        >
          x
        </button>
      </div>
      <img
        alt="a pokemon character"
        src="/test.svg"
      />
      <p>
        <b>
          Height:
        </b>
         
        30
         (decimetre)
      </p>
      <p>
        <b>
          Weight:
        </b>
         
        99
         (hectogram)
      </p>
      <p>
        <b>
          Abilities:
        </b>
        <button>
          solar-flaire
        </button>
      </p>
      <p>
        <b>
          Types:
        </b>
        <button>
          wild
        </button>
      </p>
    </div>
  </div>
</div>
`;
